File: tools/bench/answers/go_haiku/s2/04_binary.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{13, 64}
	for _, n := range nums {
		var binary string
		if n == 0 {
			binary = "0"
		} else {
			for n > 0 {
				if n%2 == 0 {
					binary = "0" + binary
				} else {
					binary = "1" + binary
				}
				n /= 2
			}
		}
		fmt.Println(binary)
	}
}

File: tools/bench/answers/rust_haiku/s3/04_binary.rs
fn main() {
    let nums = [13, 64];

    for &n in &nums {
        let mut num = n;
        let mut binary = String::new();

        if num == 0 {
            binary.push('0');
        } else {
            while num > 0 {
                if num % 2 == 1 {
                    binary.push('1');
                } else {
                    binary.push('0');
                }
                num /= 2;
            }
            binary = binary.chars().rev().collect();
        }

        println!("{}", binary);
    }
}
